fw/application/src/mui/mui_canvas.c

```diff
@@ -59,6 +59,33 @@ uint8_t mui_canvas_draw_glyph(mui_canvas_t *p_canvas, uint8_t x, uint8_t y, uint
     u8g2_DrawGlyph(p_canvas->fb, x, y, encoding);
 }
 
+uint16_t mui_canvas_draw_utf8_truncate(mui_canvas_t* p_canvas, uint8_t x, uint8_t y, uint8_t max_width, const char *str) {
+    char *p = str;
+    char utf8[5];
+
+    if (max_width < mui_canvas_get_utf8_width(p_canvas, str)) {
+        while (*p != 0) {
+            uint8_t utf8_size = mui_canvas_get_utf8_bytes(p);
+            memcpy(utf8, p, utf8_size);
+            utf8[utf8_size] = '\0';
+            if (x >= 0 && x <= max_width - mui_canvas_get_utf8_width(p_canvas, getLangString(_L_ELLIPSIS))) {
```

Comment:
call to mui_canvas_get_utf8_width(p_canvas, getLangString(_L_ELLIPSIS)  can be optimazed to call one and save to a local variable here.